### PR TITLE
feat: add response and json response body processor

### LINKED ISSUE #480: ResponseArgs not defined but used in JSON response processor
There is a variable `ResponseArgs` but doesn't seem to be any corresponding collection.

https://github.com/corazawaf/coraza/search?q=responseargs

But it is referenced in the JSON body processor and presumably isn't causing a nil exception because the test cases don't deal with JSON responses (?). What should be the correct behavior?

### PR DESCRIPTION
**Pull request by jptosso**
Closes #480

### CODE REVIEW

**Comment by jptosso** [comment]
I still have to implement forceResponseBody

File: internal/corazawaf/transaction.go [hunk 1]
```diff
@@ -1151,14 +1150,25 @@ func (tx *Transaction) ProcessResponseBody() (*types.Interruption, error) {
...
+		if err := b.ProcessResponse(reader, tx.Variables(), bodyprocessors.Options{}); err != nil {
+			tx.generateResbodyError(err)
+		}
+	} else {
+		buf := new(strings.Builder)
+		length, err := io.Copy(buf, reader)
+		if err != nil {
+			return tx.interruption, err
+		}
+		tx.variables.responseContentLength.Set(strconv.FormatInt(length, 10))
```
Comment (jcchavezs) [inline internal/corazawaf/transaction.go:1169]: just naive question: is this right? consider the body buffered by coraza is limited.

File: bodyprocessors/json.go [hunk 2]
```diff
@@ -32,6 +32,14 @@ func (js *jsonBodyProcessor) ProcessRequest(reader io.Reader, v rules.Transactio
 }
 
 func (js *jsonBodyProcessor) ProcessResponse(reader io.Reader, v rules.TransactionVariables, _ Options) error {
```
Comment (jcchavezs) [inline bodyprocessors/json.go:34]: We need a test for this.

File: actions/ctl.go [hunk 3]
```diff
@@ -163,6 +164,15 @@ func (a *ctlFn) Evaluate(r rules.RuleMetadata, txS rules.TransactionState) {
 		}
 	case ctlRequestBodyProcessor:
 		tx.Variables().RequestBodyProcessor().Set(strings.ToUpper(a.value))
+	case ctlResponseBodyProcessor:
+		if tx.LastPhase() <= types.PhaseResponseHeaders {
+			// We are still in tome to set the response body processor
```
Comment (jcchavezs) [inline actions/ctl.go:169]: Ping @jptosso @anuraaga

Comment (jcchavezs) [inline actions/ctl.go:169]:
```suggestion
			// We are still in time to set the response body processor
```

File: testing/engine/json.go [hunk 4]
```diff
@@ -58,7 +58,7 @@ SecRequestBodyAccess On
 SecResponseBodyAccess On
 SecResponseBodyMimeType application/json
 SecRule REQUEST_HEADERS:content-type "application/json" "id: 100, phase:1, pass, log, ctl:requestBodyProcessor=JSON"
-SecRule RESPONSE_HEADERS:content-type "application/json" "id:1000, phase:4, pass, log, ctl:responseBodyProcessor=JSON"
+SecRule RESPONSE_HEADERS:content-type "application/json" "id:1000, phase:3, pass, log, ctl:responseBodyProcessor=JSON"
```
Comment (jcchavezs) [inline testing/engine/json.go:61]: At this phase (4) we already pulled the body processors, excluding this one.

**Comment by jcchavezs** [comment]
At some point we need to do the JSON as lazy load maybe to avoid parsing the entire JSON if not using all fields.

**Comment by jcchavezs** [comment]
Let's first merge https://github.com/corazawaf/coraza/pull/679

**Pull request closed without merging**